Syncer container in CrashLoopBackoff

### What happened?

As soon we applied a PersistentVolume, we got the 

> "panic: runtime error: invalid memory address or nil pointer dereference"  

error from the syncer.

Here full logs 
```
I0906 12:42:11.655918       1 start.go:297] Start Plugins Manager...
I0906 12:42:11.657418       1 plugin.go:164] Plugin server listening on localhost:10099
I0906 12:42:11.658043       1 start.go:309] Using physical cluster at https://10.96.0.1:443
I0906 12:42:12.883267       1 start.go:340] Can connect to virtual cluster with version v1.23.7+k3s1
I0906 12:42:13.085295       1 plugins.go:158] Loaded 1 mutating admission controller(s) successfully in the following order: MutatingAdmissionWebhook.
I0906 12:42:13.085317       1 plugins.go:161] Loaded 1 validating admission controller(s) successfully in the following order: ValidatingAdmissionWebhook.
I0906 12:42:13.085667       1 register.go:79] Start ingresses sync controller
I0906 12:42:13.095311       1 register.go:79] Start configmaps sync controller
I0906 12:42:13.096226       1 register.go:79] Start secrets sync controller
I0906 12:42:13.100279       1 register.go:79] Start networkpolicies sync controller
I0906 12:42:13.100327       1 register.go:79] Start pods sync controller
I0906 12:42:13.100651       1 register.go:79] Start nodes sync controller
I0906 12:42:13.100797       1 register.go:79] Start persistentvolumes sync controller
I0906 12:42:13.100840       1 register.go:79] Start endpoints sync controller
I0906 12:42:13.100882       1 register.go:79] Start events sync controller
I0906 12:42:13.100907       1 register.go:79] Start legacy-storageclasses sync controller
I0906 12:42:13.100931       1 register.go:79] Start services sync controller
I0906 12:42:13.100953       1 register.go:79] Start persistentvolumeclaims sync controller
serviceaccount/coredns unchanged
clusterrole.rbac.authorization.k8s.io/system:coredns unchanged
clusterrolebinding.rbac.authorization.k8s.io/system:coredns unchanged
configmap/coredns configured
deployment.apps/coredns configured
service/kube-dns unchanged
I0906 12:42:13.192582       1 start.go:404] CoreDNS configuration from the manifest file applied successfully
I0906 12:42:13.306213       1 controller.go:185] controller coredns_nodehosts controllerGroup  controllerKind ConfigMap: Starting EventSource source kind source: *v1.ConfigMap
I0906 12:42:13.306289       1 controller.go:185] controller coredns_nodehosts controllerGroup  controllerKind ConfigMap: Starting EventSource source kind source: *v1.Node
I0906 12:42:13.306313       1 controller.go:193] controller coredns_nodehosts controllerGroup  controllerKind ConfigMap: Starting Controller
I0906 12:42:13.306474       1 controller.go:193] controller init_manifests controllerGroup  controllerKind ConfigMap: Starting Controller
I0906 12:42:13.306503       1 controller.go:227] controller init_manifests controllerGroup  controllerKind ConfigMap: Starting workers worker count 1
I0906 12:42:13.309733       1 controller.go:185] controller kubernetes_default_endpoint controllerGroup  controllerKind Endpoints: Starting EventSource source kind source: *v1.Endpoints
I0906 12:42:13.309874       1 controller.go:185] controller kubernetes_default_endpoint controllerGroup  controllerKind Endpoints: Starting EventSource source &{{%!s(*v1.Endpoints=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} []}) %!s(*cache.informerCache=&{0xc0003cd580}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 12:42:13.309920       1 controller.go:185] controller kubernetes_default_endpoint controllerGroup  controllerKind Endpoints: Starting EventSource source &{{%!s(*v1.EndpointSlice=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []}  [] []}) %!s(*cache.informerCache=&{0xc0003cd580}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 12:42:13.309943       1 controller.go:193] controller kubernetes_default_endpoint controllerGroup  controllerKind Endpoints: Starting Controller
I0906 12:42:13.310363       1 controller.go:150] controller init_manifests controllerGroup  controllerKind ConfigMap: Starting EventSource source &source.Kind{Type:(*v1.ConfigMap)(0xc000d9af00), cache:(*cache.informerCache)(0xc000116490), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 12:42:13.310621       1 controller.go:193] controller configmap controllerGroup  controllerKind ConfigMap: Starting Controller
I0906 12:42:13.310652       1 controller.go:227] controller configmap controllerGroup  controllerKind ConfigMap: Starting workers worker count 1
I0906 12:42:13.321894       1 controller.go:150] controller configmap controllerGroup  controllerKind ConfigMap: Starting EventSource source &source.Kind{Type:(*v1.ConfigMap)(0xc000d9b180), cache:(*cache.informerCache)(0xc0002d26e8), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 12:42:13.321968       1 controller.go:150] controller configmap controllerGroup  controllerKind ConfigMap: Starting EventSource source &source.kindWithCache{kind:source.Kind{Type:(*v1.ConfigMap)(0xc000d9b040), cache:(*cache.informerCache)(0xc000116490), started:(chan error)(nil), startCancel:(func())(nil)}}
I0906 12:42:13.322012       1 controller.go:150] controller configmap controllerGroup  controllerKind ConfigMap: Starting EventSource source &source.Kind{Type:(*v1.Pod)(0xc000e2b000), cache:(*cache.informerCache)(0xc0002d26e8), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 12:42:13.322149       1 controller.go:193] controller networkpolicy controllerGroup networking.k8s.io controllerKind NetworkPolicy: Starting Controller
I0906 12:42:13.322181       1 controller.go:227] controller networkpolicy controllerGroup networking.k8s.io controllerKind NetworkPolicy: Starting workers worker count 1
I0906 12:42:13.322660       1 controller.go:185] controller ingress controllerGroup networking.k8s.io controllerKind Ingress: Starting EventSource source kind source: *v1.Ingress
I0906 12:42:13.322727       1 controller.go:185] controller ingress controllerGroup networking.k8s.io controllerKind Ingress: Starting EventSource source &{{%!s(*v1.Ingress=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} {<nil> <nil> [] []} {{[]}}}) %!s(*cache.informerCache=&{0xc00012dd00}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 12:42:13.322774       1 controller.go:193] controller ingress controllerGroup networking.k8s.io controllerKind Ingress: Starting Controller
I0906 12:42:13.323131       1 controller.go:185] controller secret controllerGroup  controllerKind Secret: Starting EventSource source kind source: *v1.Secret
I0906 12:42:13.323200       1 controller.go:185] controller secret controllerGroup  controllerKind Secret: Starting EventSource source &{{%!s(*v1.Secret=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} <nil> map[] map[] }) %!s(*cache.informerCache=&{0xc00012dd00}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 12:42:13.323232       1 controller.go:185] controller secret controllerGroup  controllerKind Secret: Starting EventSource source kind source: *v1.Ingress
I0906 12:42:13.323265       1 controller.go:185] controller secret controllerGroup  controllerKind Secret: Starting EventSource source kind source: *v1.Pod
I0906 12:42:13.323286       1 controller.go:193] controller secret controllerGroup  controllerKind Secret: Starting Controller
I0906 12:42:13.323374       1 controller.go:150] controller networkpolicy controllerGroup networking.k8s.io controllerKind NetworkPolicy: Starting EventSource source &source.Kind{Type:(*v1.NetworkPolicy)(0xc00144a820), cache:(*cache.informerCache)(0xc0002d26e8), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 12:42:13.323419       1 controller.go:150] controller networkpolicy controllerGroup networking.k8s.io controllerKind NetworkPolicy: Starting EventSource source &source.kindWithCache{kind:source.Kind{Type:(*v1.NetworkPolicy)(0xc00144a680), cache:(*cache.informerCache)(0xc000116490), started:(chan error)(nil), startCancel:(func())(nil)}}
I0906 12:42:13.328375       1 controller.go:185] controller pod controllerGroup  controllerKind Pod: Starting EventSource source kind source: *v1.Pod
I0906 12:42:13.328452       1 controller.go:185] controller pod controllerGroup  controllerKind Pod: Starting EventSource source &{{%!s(*v1.Pod=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} {[] [] [] []  <nil> <nil>  map[]   <nil>  false false false <nil> <nil> []   <nil>  [] []  <nil> <nil> [] <nil> <nil> <nil> map[] [] <nil> <nil>} { []      [] <nil> [] []  []}}) %!s(*cache.informerCache=&{0xc00012dd00}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 12:42:13.328486       1 controller.go:185] controller pod controllerGroup  controllerKind Pod: Starting EventSource source kind source: *v1.Namespace
I0906 12:42:13.328514       1 controller.go:193] controller pod controllerGroup  controllerKind Pod: Starting Controller
I0906 12:42:13.331116       1 controller.go:193] controller persistentvolume controllerGroup  controllerKind PersistentVolume: Starting Controller
I0906 12:42:13.331133       1 controller.go:227] controller persistentvolume controllerGroup  controllerKind PersistentVolume: Starting workers worker count 1
I0906 12:42:13.331160       1 controller.go:150] controller persistentvolume controllerGroup  controllerKind PersistentVolume: Starting EventSource source &source.Kind{Type:(*v1.PersistentVolume)(0xc0013fdb80), cache:(*cache.informerCache)(0xc0002d26e8), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 12:42:13.331179       1 controller.go:150] controller persistentvolume controllerGroup  controllerKind PersistentVolume: Starting EventSource source &source.kindWithCache{kind:source.Kind{Type:(*v1.PersistentVolume)(0xc0013fd900), cache:(*cache.informerCache)(0xc000116490), started:(chan error)(nil), startCancel:(func())(nil)}}
I0906 12:42:13.331216       1 controller.go:150] controller persistentvolume controllerGroup  controllerKind PersistentVolume: Starting EventSource source &source.Kind{Type:(*v1.PersistentVolumeClaim)(0xc000af43c0), cache:(*cache.informerCache)(0xc0002d26e8), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 12:42:13.331297       1 controller.go:193] controller event controllerGroup  controllerKind Event: Starting Controller
I0906 12:42:13.331310       1 controller.go:227] controller event controllerGroup  controllerKind Event: Starting workers worker count 1
I0906 12:42:13.331347       1 controller.go:185] controller node controllerGroup  controllerKind Node: Starting EventSource source kind source: *v1.Node
I0906 12:42:13.331384       1 controller.go:185] controller node controllerGroup  controllerKind Node: Starting EventSource source &{{%!s(*v1.Node=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} { []  false [] <nil> } {map[] map[]  [] [] {{0}} {         } [] [] [] <nil>}}) %!s(*cache.informerCache=&{0xc00012dd00}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 12:42:13.331408       1 controller.go:185] controller node controllerGroup  controllerKind Node: Starting EventSource source &{{%!s(*v1.Pod=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} {[] [] [] []  <nil> <nil>  map[]   <nil>  false false false <nil> <nil> []   <nil>  [] []  <nil> <nil> [] <nil> <nil> <nil> map[] [] <nil> <nil>} { []      [] <nil> [] []  []}}) %!s(*cache.informerCache=&{0xc00012dd00}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 12:42:13.331421       1 controller.go:185] controller node controllerGroup  controllerKind Node: Starting EventSource source kind source: *v1.Pod
I0906 12:42:13.331432       1 controller.go:193] controller node controllerGroup  controllerKind Node: Starting Controller
I0906 12:42:13.331690       1 controller.go:185] controller endpoints controllerGroup  controllerKind Endpoints: Starting EventSource source kind source: *v1.Endpoints
I0906 12:42:13.331726       1 controller.go:185] controller endpoints controllerGroup  controllerKind Endpoints: Starting EventSource source &{{%!s(*v1.Endpoints=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} []}) %!s(*cache.informerCache=&{0xc00012dd00}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 12:42:13.347707       1 controller.go:193] controller endpoints controllerGroup  controllerKind Endpoints: Starting Controller
I0906 12:42:13.347833       1 controller.go:150] controller event controllerGroup  controllerKind Event: Starting EventSource source &source.Kind{Type:(*v1.Event)(0xc00084f400), cache:(*cache.informerCache)(0xc0002d26e8), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 12:42:13.347873       1 controller.go:150] controller event controllerGroup  controllerKind Event: Starting EventSource source &source.kindWithCache{kind:source.Kind{Type:(*v1.Event)(0xc00084f180), cache:(*cache.informerCache)(0xc000116490), started:(chan error)(nil), startCancel:(func())(nil)}}
I0906 12:42:13.347967       1 controller.go:193] controller service controllerGroup  controllerKind Service: Starting Controller
I0906 12:42:13.347982       1 controller.go:227] controller service controllerGroup  controllerKind Service: Starting workers worker count 1
I0906 12:42:13.349187       1 controller.go:185] controller legacy-storageclass controllerGroup storage.k8s.io controllerKind StorageClass: Starting EventSource source kind source: *v1.StorageClass
I0906 12:42:13.349227       1 controller.go:185] controller legacy-storageclass controllerGroup storage.k8s.io controllerKind StorageClass: Starting EventSource source &{{%!s(*v1.StorageClass=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []}  map[] <nil> [] <nil> <nil> []}) %!s(*cache.informerCache=&{0xc00012dd00}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 12:42:13.349235       1 controller.go:193] controller legacy-storageclass controllerGroup storage.k8s.io controllerKind StorageClass: Starting Controller
I0906 12:42:13.349334       1 controller.go:150] controller service controllerGroup  controllerKind Service: Starting EventSource source &source.Kind{Type:(*v1.Service)(0xc00084f900), cache:(*cache.informerCache)(0xc0002d26e8), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 12:42:13.349357       1 controller.go:150] controller service controllerGroup  controllerKind Service: Starting EventSource source &source.kindWithCache{kind:source.Kind{Type:(*v1.Service)(0xc00084f680), cache:(*cache.informerCache)(0xc000116490), started:(chan error)(nil), startCancel:(func())(nil)}}
I0906 12:42:13.349427       1 controller.go:185] controller persistent-volume-claim controllerGroup  controllerKind PersistentVolumeClaim: Starting EventSource source kind source: *v1.PersistentVolumeClaim
I0906 12:42:13.349463       1 controller.go:185] controller persistent-volume-claim controllerGroup  controllerKind PersistentVolumeClaim: Starting EventSource source &{{%!s(*v1.PersistentVolumeClaim=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} {[] <nil> {map[] map[]}  <nil> <nil> <nil> <nil>} { [] map[] [] map[] <nil>}}) %!s(*cache.informerCache=&{0xc00012dd00}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 12:42:13.349476       1 controller.go:193] controller persistent-volume-claim controllerGroup  controllerKind PersistentVolumeClaim: Starting Controller
I0906 12:42:13.406758       1 controller.go:227] controller coredns_nodehosts controllerGroup  controllerKind ConfigMap: Starting workers worker count 1
I0906 12:42:13.414482       1 nodehosts.go:57] corednsnodehosts-controller: CoreDNS ConfigMap CreateOrPatch operation result: updated
I0906 12:42:13.423123       1 controller.go:227] controller kubernetes_default_endpoint controllerGroup  controllerKind Endpoints: Starting workers worker count 1
I0906 12:42:13.423939       1 init.go:62] init-manifests-controller: process init manifests
I0906 12:42:13.425509       1 controller.go:227] controller ingress controllerGroup networking.k8s.io controllerKind Ingress: Starting workers worker count 1
I0906 12:42:13.427915       1 controller.go:227] controller secret controllerGroup  controllerKind Secret: Starting workers worker count 1
I0906 12:42:13.448914       1 controller.go:117] controller persistentvolume controllerGroup  controllerKind PersistentVolume: persistentVolume klog.ObjectRef{Name:"my-volume", Namespace:""} namespace  name my-volume: reconcileID "c3a78aae-eb85-417a-a54d-8f2678da7589": : Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x18843ca]

goroutine 797 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x1ee7cc0, 0x36bb760})
	/usr/local/go/src/runtime/panic.go:838 +0x207
github.com/loft-sh/vcluster/pkg/controllers/resources/persistentvolumes.(*persistentVolumeSyncer).translateUpdateBackwards(0xc0005dbf20?, 0xc000010150, 0xc00125c000, 0xc00125c780, 0x0)
	/vcluster-dev/pkg/controllers/resources/persistentvolumes/translate.go:70 +0x22a
github.com/loft-sh/vcluster/pkg/controllers/resources/persistentvolumes.(*persistentVolumeSyncer).Sync(0x1ec6880?, 0xc000010150, {0x25939f0?, 0xc00125c780}, {0x25939f0, 0xc00125c000})
	/vcluster-dev/pkg/controllers/resources/persistentvolumes/syncer.go:153 +0x447
github.com/loft-sh/vcluster/pkg/controllers/syncer.(*syncerController).Reconcile(0xc000775500, {0x2580690?, 0xc001252c00}, {{{0x0?, 0x10?}, {0xc0008a26d0?, 0x40d9a7?}}})
	/vcluster-dev/pkg/controllers/syncer/syncer.go:109 +0x705
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x25805e8?, {0x2580690?, 0xc001252c00?}, {{{0x0?, 0x2106f80?}, {0xc0008a26d0?, 0x4041f4?}}})
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:121 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0002af680, {0x25805e8, 0xc000b4d1c0}, {0x1fa4960?, 0xc000afde80?})
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:320 +0x33c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0002af680, {0x25805e8, 0xc000b4d1c0})
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:273 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:234 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:230 +0x32
```

### What did you expect to happen?

We expected that the PV was applied in the vcluster. We can't access the vcluster right now beacuse the syncer is in crashbackoff state hence it's not responding.

### How can we reproduce it (as minimally and precisely as possible)?

This is the Vcluster manifest:

```
---
# Source: vcluster/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: vc-devops
  namespace: devops
  labels:
    app: vcluster
    chart: "vcluster-0.11.2"
    release: "devops"
    heritage: "Helm"

---
# Source: vcluster/templates/workloadserviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: vc-workload-devops
  namespace: devops
  labels:
    app: vcluster
    chart: "vcluster-0.11.2"
    release: "devops"
    heritage: "Helm"

---
# Source: vcluster/templates/coredns.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: devops-coredns
  namespace: devops
data:
  coredns.yaml: |-
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      name: coredns
      namespace: kube-system
    ---
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        kubernetes.io/bootstrapping: rbac-defaults
      name: system:coredns
    rules:
      - apiGroups:
          - ""
        resources:
          - endpoints
          - services
          - pods
          - namespaces
        verbs:
          - list
          - watch
      - apiGroups:
          - discovery.k8s.io
        resources:
          - endpointslices
        verbs:
          - list
          - watch
    ---
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      annotations:
        rbac.authorization.kubernetes.io/autoupdate: "true"
      labels:
        kubernetes.io/bootstrapping: rbac-defaults
      name: system:coredns
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: system:coredns
    subjects:
      - kind: ServiceAccount
        name: coredns
        namespace: kube-system
    ---
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: coredns
      namespace: kube-system
    data:
      Corefile: |
        .:1053 {
            {{.LOG_IN_DEBUG}}
            errors
            health
            ready
            kubernetes cluster.local in-addr.arpa ip6.arpa {
              pods insecure
              fallthrough in-addr.arpa ip6.arpa
            }
            hosts /etc/coredns/NodeHosts {
              ttl 60
              reload 15s
              fallthrough
            }
            prometheus :9153
            forward . /etc/resolv.conf
            cache 30
            loop
            reload
            loadbalance
        }

        import /etc/coredns/custom/*.server
      NodeHosts: ""
    ---
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: coredns
      namespace: kube-system
      labels:
        k8s-app: kube-dns
        kubernetes.io/name: "CoreDNS"
    spec:
      replicas: 1
      strategy:
        type: RollingUpdate
        rollingUpdate:
          maxUnavailable: 1
      selector:
        matchLabels:
          k8s-app: kube-dns
      template:
        metadata:
          labels:
            k8s-app: kube-dns
        spec:
          priorityClassName: "system-cluster-critical"
          serviceAccountName: coredns
          nodeSelector:
            kubernetes.io/os: linux
          topologySpreadConstraints:
            - maxSkew: 1
              topologyKey: kubernetes.io/hostname
              whenUnsatisfiable: DoNotSchedule
              labelSelector:
                matchLabels:
                  k8s-app: kube-dns
          securityContext:
            seccompProfile:
              type: RuntimeDefault
          containers:
            - name: coredns
              image: {{.IMAGE}}
              imagePullPolicy: IfNotPresent
              resources:
                limits:
                  cpu: 1000m
                  memory: 170Mi
                requests:
                  cpu: 3m
                  memory: 16Mi
              args: [ "-conf", "/etc/coredns/Corefile" ]
              volumeMounts:
                - name: config-volume
                  mountPath: /etc/coredns
                  readOnly: true
                - name: custom-config-volume
                  mountPath: /etc/coredns/custom
                  readOnly: true
              ports:
                - containerPort: 1053
                  name: dns
                  protocol: UDP
                - containerPort: 1053
                  name: dns-tcp
                  protocol: TCP
                - containerPort: 9153
                  name: metrics
                  protocol: TCP
              securityContext:
                runAsUser: {{.RUN_AS_USER}}
                runAsNonRoot: {{.RUN_AS_NON_ROOT}}
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
              livenessProbe:
                httpGet:
                  path: /health
                  port: 8080
                  scheme: HTTP
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 1
                successThreshold: 1
                failureThreshold: 3
              readinessProbe:
                httpGet:
                  path: /ready
                  port: 8181
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 2
                timeoutSeconds: 1
                successThreshold: 1
                failureThreshold: 3
          dnsPolicy: Default
          volumes:
            - name: config-volume
              configMap:
                name: coredns
                items:
                  - key: Corefile
                    path: Corefile
                  - key: NodeHosts
                    path: NodeHosts
            - name: custom-config-volume
              configMap:
                name: coredns-custom
                optional: true
    ---
    apiVersion: v1
    kind: Service
    metadata:
      name: kube-dns
      namespace: kube-system
      annotations:
        prometheus.io/port: "9153"
        prometheus.io/scrape: "true"
      labels:
        k8s-app: kube-dns
        kubernetes.io/cluster-service: "true"
        kubernetes.io/name: "CoreDNS"
    spec:
      selector:
        k8s-app: kube-dns
      type: ClusterIP
      ports:
        - name: dns
          port: 53
          targetPort: 1053
          protocol: UDP
        - name: dns-tcp
          port: 53
          targetPort: 1053
          protocol: TCP
        - name: metrics
          port: 9153
          protocol: TCP
---
# Source: vcluster/templates/init-configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: devops-init-manifests
  namespace: devops
  labels:
    app: vcluster
    chart: "vcluster-0.11.2"
    release: "devops"
    heritage: "Helm"
data:
  initmanifests.yaml: |-
    ---
---
# Source: vcluster/templates/rbac/clusterrole.yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: vc-devops-v-devops
  labels:
    app: vcluster
    chart: "vcluster-0.11.2"
    release: "devops"
    heritage: "Helm"
rules:
  - apiGroups: [""]
    resources: ["nodes", "nodes/status"]
    verbs: ["get", "watch", "list"]
  - apiGroups: [""]
    resources: ["pods", "nodes/proxy", "nodes/metrics", "nodes/stats"]
    verbs: ["get", "watch", "list"]
  - apiGroups: [""]
    resources: ["persistentvolumes"]
    verbs: ["create", "delete", "patch", "update", "get", "watch", "list"]
  - apiGroups: ["storage.k8s.io"]
    resources: ["storageclasses"]
    verbs: ["get", "watch", "list"]
  - apiGroups: ["authentication.k8s.io"]
    resources: ["tokenreviews"]
    verbs: ["create", "delete", "patch", "update", "get", "watch", "list"]
---
# Source: vcluster/templates/rbac/clusterrolebinding.yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: vc-devops-v-devops
  labels:
    app: vcluster
    chart: "vcluster-0.11.2"
    release: "devops"
    heritage: "Helm"
subjects:
  - kind: ServiceAccount
    name: vc-devops
    namespace: devops
roleRef:
  kind: ClusterRole
  name: vc-devops-v-devops
  apiGroup: rbac.authorization.k8s.io
---
# Source: vcluster/templates/rbac/role.yaml
kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: devops
  namespace: devops
  labels:
    app: vcluster
    chart: "vcluster-0.11.2"
    release: "devops"
    heritage: "Helm"
rules:
  - apiGroups: [""]
    resources: ["configmaps", "secrets", "services", "pods", "pods/attach", "pods/portforward", "pods/exec", "persistentvolumeclaims"]
    verbs: ["create", "delete", "patch", "update", "get", "list", "watch"]
  - apiGroups: [""]
    resources: ["endpoints"]
    verbs: ["create", "delete", "patch", "update"]
  - apiGroups: [""]
    resources: ["endpoints", "events", "pods/log"]
    verbs: ["get", "list", "watch"]
  - apiGroups: ["networking.k8s.io"]
    resources: ["ingresses"]
    verbs: ["create", "delete", "patch", "update", "get", "list", "watch"]
  - apiGroups: ["apps"]
    resources: ["statefulsets", "replicasets", "deployments"]
    verbs: ["get", "list", "watch"]
  - apiGroups: ["networking.k8s.io"]
    resources: ["networkpolicies"]
    verbs: ["create", "delete", "patch", "update", "get", "list", "watch"]
---
# Source: vcluster/templates/rbac/rolebinding.yaml
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: devops
  namespace: devops
  labels:
    app: vcluster
    chart: "vcluster-0.11.2"
    release: "devops"
    heritage: "Helm"
subjects:
  - kind: ServiceAccount
    name: vc-devops
    namespace: devops
roleRef:
  kind: Role
  name: devops
  apiGroup: rbac.authorization.k8s.io
---
# Source: vcluster/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: devops
  namespace: devops
  labels:
    app: vcluster
    chart: "vcluster-0.11.2"
    release: "devops"
    heritage: "Helm"
spec:
  type: ClusterIP
  ports:
    - name: https
      port: 443
      targetPort: 8443
      protocol: TCP
    - name: k3s-metrics
      port: 6444
      targetPort: 6444
      protocol: TCP
  selector:
    app: vcluster
    release: devops
---
# Source: vcluster/templates/statefulset-service.yaml
apiVersion: v1
kind: Service
metadata:
  name: devops-headless
  namespace: devops
  labels:
    app: devops-vcluster
    chart: "vcluster-0.11.2"
    release: "devops"
    heritage: "Helm"
spec:
  ports:
    - name: https
      port: 443
      targetPort: 8443
      protocol: TCP
  clusterIP: None
  selector:
    app: vcluster
    release: "devops"
---
# Source: vcluster/templates/statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: devops
  namespace: devops
  labels:
    app: vcluster
    chart: "vcluster-0.11.2"
    release: "devops"
    heritage: "Helm"
spec:
  serviceName: devops-headless
  replicas: 1
  selector:
    matchLabels:
      app: vcluster
      release: devops
  volumeClaimTemplates:
    - metadata:
        name: data
      spec:
        accessModes: [ "ReadWriteOnce" ]
        storageClassName: oci-bv
        resources:
          requests:
            storage: 50Gi
  template:
    metadata:
      labels:
        app: vcluster
        release: devops
    spec:
      terminationGracePeriodSeconds: 10
      affinity:
        {}
      tolerations:
        - key: "workload"
          operator: "Equal"
          value: "critical"
          effect: "NoSchedule"
        - key: "team"
          operator: "Equal"
          value: "devops"
          effect: "NoSchedule"
      nodeSelector:
        workload: critical
        team: devops
      serviceAccountName: vc-devops
      volumes:
        - name: config
          emptyDir: {}
        - name: dex-openid-certs
          secret:
            secretName: dex-openid-certs
        - name: coredns
          configMap:
            name: devops-coredns
      imagePullSecrets:
        - name: regcred
      containers:
      - image: myregistry/rancher-k3s:release-v1.23.7-k3s1
        name: vcluster
        # k3s has a problem running as pid 1 and disabled agents on cgroupv2
        # nodes as it will try to evacuate the cgroups there. Starting k3s
        # through a shell makes it non pid 1 and prevents this from happening
        command:
          - /bin/sh
        args:
          - -c
          - /bin/k3s
            server
            --write-kubeconfig=/data/k3s-config/kube-config.yaml
            --data-dir=/data
            --disable=traefik,servicelb,metrics-server,local-storage,coredns
            --disable-network-policy
            --disable-agent
            --disable-cloud-controller
            --flannel-backend=none
            --disable-scheduler
            --kube-controller-manager-arg=controllers=*,-nodeipam,-nodelifecycle,-persistentvolume-binder,-attachdetach,-persistentvolume-expander,-cloud-node-lifecycle
            --service-cidr=10.96.0.0/16
            --kube-apiserver-arg=endpoint-reconciler-type=none
            --kube-apiserver-arg oidc-issuer-url=$OIDC_ISSUER_URL
            --kube-apiserver-arg oidc-client-id=kubernetes
            --kube-apiserver-arg oidc-ca-file=/var/openid-ca.pem
            --kube-apiserver-arg oidc-username-claim=email
            --kube-apiserver-arg oidc-groups-claim=groups
            && true
        env:
          - name: OIDC_ISSUER_URL 
            valueFrom:
              configMapKeyRef:
                name: oidc-issuer-url
                key: oidc_issuer_url
        securityContext:
          allowPrivilegeEscalation: false
        volumeMounts:
          - name: config
            mountPath: /etc/rancher
          - mountPath: /data
            name: data
          - mountPath: /var
            name: dex-openid-certs
        resources:
          limits:
            memory: 2Gi
          requests:
            cpu: 200m
            memory: 256Mi
      - name: syncer
        image: "myregistry/loftsh/vcluster:release-0.11.2"
        args:
          - --name=devops
          - --service-account=vc-workload-devops  
          #- --enforce-pod-security-standard=privileged   
          - --sync=-fake-persistentvolumes
          - --sync=legacy-storageclasses
          - --sync=networkpolicies
          - --sync=persistentvolumes
          - --sync=nodes
          - "--tls-san=mysan.it"
          - "--override-hosts-container-image=alpine:release-3.13.1"                    
        livenessProbe:
          httpGet:
            path: /healthz
            port: 8443
            scheme: HTTPS
          failureThreshold: 60
          initialDelaySeconds: 60
          periodSeconds: 2
        readinessProbe:
          httpGet:
            path: /readyz
            port: 8443
            scheme: HTTPS
          failureThreshold: 60
          periodSeconds: 2
        securityContext:
          allowPrivilegeEscalation: false
        env:
          - name: POD_IP
            valueFrom:
              fieldRef:
                fieldPath: status.podIP
          - name: VCLUSTER_NODE_NAME
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
        volumeMounts:
          - name: coredns
            mountPath: /manifests/coredns
            readOnly: true
          - mountPath: /data
            name: data
            readOnly: true
        resources:
          limits:
            memory: 1Gi
          requests:
            cpu: 10m
            memory: 64Mi
```

### Anything else we need to know?

We were able to recreated the scenario (and trigger than the crashloopbackoff) by bringing up a local cluster with Kind and deploying the same vcluster manifest. Then we just applied a PersistentVolume.

PV Manifest
```
apiVersion: v1
kind: PersistentVolume
metadata:
  name: myvolume
  namespace: pam-pipelines
spec:
  storageClassName: standard
  accessModes:
    - ReadWriteOnce
  capacity:
    storage: 1Gi
  local:
    path: "/Users/polidori/Documents/Demo"
  nodeAffinity:
    required:
      nodeSelectorTerms:
      - matchExpressions:
        - key: kubernetes.io/hostname
          operator: In
          values:
          - kind-control-plane
```

Logs from the syncer:
```
I0906 13:24:38.935266       1 start.go:297] Start Plugins Manager...
I0906 13:24:38.937849       1 plugin.go:164] Plugin server listening on localhost:10099
I0906 13:24:38.938616       1 start.go:309] Using physical cluster at https://10.96.0.1:443
I0906 13:24:39.237402       1 start.go:340] Can connect to virtual cluster with version v1.23.7+k3s1
I0906 13:24:39.541891       1 plugins.go:158] Loaded 1 mutating admission controller(s) successfully in the following order: MutatingAdmissionWebhook.
I0906 13:24:39.542561       1 plugins.go:161] Loaded 1 validating admission controller(s) successfully in the following order: ValidatingAdmissionWebhook.
I0906 13:24:39.543151       1 register.go:79] Start configmaps sync controller
I0906 13:24:39.546324       1 register.go:79] Start secrets sync controller
I0906 13:24:39.590701       1 register.go:79] Start pods sync controller
I0906 13:24:39.591752       1 register.go:79] Start legacy-storageclasses sync controller
I0906 13:24:39.592676       1 register.go:79] Start persistentvolumes sync controller
I0906 13:24:39.593275       1 register.go:79] Start services sync controller
I0906 13:24:39.593608       1 register.go:79] Start events sync controller
I0906 13:24:39.593951       1 register.go:79] Start nodes sync controller
I0906 13:24:39.595524       1 register.go:79] Start endpoints sync controller
I0906 13:24:39.595569       1 register.go:79] Start persistentvolumeclaims sync controller
I0906 13:24:39.595606       1 register.go:79] Start ingresses sync controller
I0906 13:24:39.623800       1 register.go:79] Start networkpolicies sync controller
serviceaccount/coredns unchanged
clusterrole.rbac.authorization.k8s.io/system:coredns unchanged
clusterrolebinding.rbac.authorization.k8s.io/system:coredns unchanged
configmap/coredns unchanged
I0906 13:24:40.417850       1 controller.go:185] controller coredns_nodehosts controllerGroup  controllerKind ConfigMap: Starting EventSource source kind source: *v1.ConfigMap
I0906 13:24:40.418295       1 controller.go:185] controller coredns_nodehosts controllerGroup  controllerKind ConfigMap: Starting EventSource source kind source: *v1.Node
I0906 13:24:40.418560       1 controller.go:193] controller coredns_nodehosts controllerGroup  controllerKind ConfigMap: Starting Controller
I0906 13:24:40.423523       1 controller.go:193] controller init_manifests controllerGroup  controllerKind ConfigMap: Starting Controller
I0906 13:24:40.424573       1 controller.go:227] controller init_manifests controllerGroup  controllerKind ConfigMap: Starting workers worker count 1
I0906 13:24:40.429418       1 controller.go:185] controller kubernetes_default_endpoint controllerGroup  controllerKind Endpoints: Starting EventSource source kind source: *v1.Endpoints
I0906 13:24:40.430470       1 controller.go:185] controller kubernetes_default_endpoint controllerGroup  controllerKind Endpoints: Starting EventSource source &{{%!s(*v1.Endpoints=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} []}) %!s(*cache.informerCache=&{0xc000089700}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 13:24:40.431003       1 controller.go:185] controller kubernetes_default_endpoint controllerGroup  controllerKind Endpoints: Starting EventSource source &{{%!s(*v1.EndpointSlice=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []}  [] []}) %!s(*cache.informerCache=&{0xc000089700}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 13:24:40.434049       1 controller.go:193] controller kubernetes_default_endpoint controllerGroup  controllerKind Endpoints: Starting Controller
I0906 13:24:40.434879       1 controller.go:150] controller init_manifests controllerGroup  controllerKind ConfigMap: Starting EventSource source &source.Kind{Type:(*v1.ConfigMap)(0xc0000e2280), cache:(*cache.informerCache)(0xc0003786d8), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 13:24:40.436047       1 controller.go:193] controller secret controllerGroup  controllerKind Secret: Starting Controller
I0906 13:24:40.436441       1 controller.go:227] controller secret controllerGroup  controllerKind Secret: Starting workers worker count 1
I0906 13:24:40.439781       1 controller.go:185] controller configmap controllerGroup  controllerKind ConfigMap: Starting EventSource source kind source: *v1.ConfigMap
I0906 13:24:40.440384       1 controller.go:185] controller configmap controllerGroup  controllerKind ConfigMap: Starting EventSource source &{{%!s(*v1.ConfigMap=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} <nil> map[] map[]}) %!s(*cache.informerCache=&{0xc00097dc40}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 13:24:40.442864       1 controller.go:185] controller configmap controllerGroup  controllerKind ConfigMap: Starting EventSource source kind source: *v1.Pod
I0906 13:24:40.443359       1 controller.go:193] controller configmap controllerGroup  controllerKind ConfigMap: Starting Controller
I0906 13:24:40.443994       1 controller.go:150] controller secret controllerGroup  controllerKind Secret: Starting EventSource source &source.Kind{Type:(*v1.Secret)(0xc0000e3900), cache:(*cache.informerCache)(0xc000378998), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 13:24:40.446238       1 controller.go:150] controller secret controllerGroup  controllerKind Secret: Starting EventSource source &source.kindWithCache{kind:source.Kind{Type:(*v1.Secret)(0xc0000e3040), cache:(*cache.informerCache)(0xc0003786d8), started:(chan error)(nil), startCancel:(func())(nil)}}
I0906 13:24:40.446961       1 controller.go:150] controller secret controllerGroup  controllerKind Secret: Starting EventSource source &source.Kind{Type:(*v1.Ingress)(0xc00073cf00), cache:(*cache.informerCache)(0xc000378998), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 13:24:40.447267       1 controller.go:150] controller secret controllerGroup  controllerKind Secret: Starting EventSource source &source.Kind{Type:(*v1.Pod)(0xc000b4b800), cache:(*cache.informerCache)(0xc000378998), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 13:24:40.447735       1 controller.go:193] controller legacy-storageclass controllerGroup storage.k8s.io controllerKind StorageClass: Starting Controller
I0906 13:24:40.453017       1 controller.go:227] controller legacy-storageclass controllerGroup storage.k8s.io controllerKind StorageClass: Starting workers worker count 1
I0906 13:24:40.475339       1 controller.go:185] controller pod controllerGroup  controllerKind Pod: Starting EventSource source kind source: *v1.Pod
I0906 13:24:40.476014       1 controller.go:185] controller pod controllerGroup  controllerKind Pod: Starting EventSource source &{{%!s(*v1.Pod=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} {[] [] [] []  <nil> <nil>  map[]   <nil>  false false false <nil> <nil> []   <nil>  [] []  <nil> <nil> [] <nil> <nil> <nil> map[] [] <nil> <nil>} { []      [] <nil> [] []  []}}) %!s(*cache.informerCache=&{0xc00097dc40}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 13:24:40.476419       1 controller.go:185] controller pod controllerGroup  controllerKind Pod: Starting EventSource source kind source: *v1.Namespace
I0906 13:24:40.476762       1 controller.go:193] controller pod controllerGroup  controllerKind Pod: Starting Controller
I0906 13:24:40.477417       1 controller.go:150] controller legacy-storageclass controllerGroup storage.k8s.io controllerKind StorageClass: Starting EventSource source &source.Kind{Type:(*v1.StorageClass)(0xc00073db00), cache:(*cache.informerCache)(0xc000378998), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 13:24:40.477742       1 controller.go:150] controller legacy-storageclass controllerGroup storage.k8s.io controllerKind StorageClass: Starting EventSource source &source.kindWithCache{kind:source.Kind{Type:(*v1.StorageClass)(0xc00073d980), cache:(*cache.informerCache)(0xc0003786d8), started:(chan error)(nil), startCancel:(func())(nil)}}
I0906 13:24:40.478190       1 controller.go:193] controller service controllerGroup  controllerKind Service: Starting Controller
I0906 13:24:40.478448       1 controller.go:227] controller service controllerGroup  controllerKind Service: Starting workers worker count 1
I0906 13:24:40.510200       1 controller.go:185] controller persistentvolume controllerGroup  controllerKind PersistentVolume: Starting EventSource source kind source: *v1.PersistentVolume
I0906 13:24:40.510747       1 controller.go:185] controller persistentvolume controllerGroup  controllerKind PersistentVolume: Starting EventSource source &{{%!s(*v1.PersistentVolume=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} {map[] {<nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil>} [] <nil>   [] <nil> <nil>} {  }}) %!s(*cache.informerCache=&{0xc00097dc40}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 13:24:40.511278       1 controller.go:185] controller persistentvolume controllerGroup  controllerKind PersistentVolume: Starting EventSource source kind source: *v1.PersistentVolumeClaim
I0906 13:24:40.511616       1 controller.go:193] controller persistentvolume controllerGroup  controllerKind PersistentVolume: Starting Controller
I0906 13:24:40.513329       1 controller.go:150] controller service controllerGroup  controllerKind Service: Starting EventSource source &source.Kind{Type:(*v1.Service)(0xc000745900), cache:(*cache.informerCache)(0xc000378998), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 13:24:40.513388       1 controller.go:150] controller service controllerGroup  controllerKind Service: Starting EventSource source &source.kindWithCache{kind:source.Kind{Type:(*v1.Service)(0xc000745680), cache:(*cache.informerCache)(0xc0003786d8), started:(chan error)(nil), startCancel:(func())(nil)}}
I0906 13:24:40.549808       1 controller.go:185] controller event controllerGroup  controllerKind Event: Starting EventSource source kind source: *v1.Event
I0906 13:24:40.557570       1 controller.go:185] controller event controllerGroup  controllerKind Event: Starting EventSource source &{{%!s(*v1.Event=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} {      }   { } {{0 0 <nil>}} {{0 0 <nil>}} 0  {{0 0 <nil>}} <nil>  <nil>  }) %!s(*cache.informerCache=&{0xc00097dc40}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 13:24:40.557987       1 controller.go:193] controller event controllerGroup  controllerKind Event: Starting Controller
I0906 13:24:40.559287       1 controller.go:193] controller endpoints controllerGroup  controllerKind Endpoints: Starting Controller
I0906 13:24:40.560636       1 controller.go:227] controller endpoints controllerGroup  controllerKind Endpoints: Starting workers worker count 1
I0906 13:24:40.612210       1 controller.go:185] controller node controllerGroup  controllerKind Node: Starting EventSource source kind source: *v1.Node
I0906 13:24:40.616773       1 controller.go:185] controller node controllerGroup  controllerKind Node: Starting EventSource source &{{%!s(*v1.Node=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} { []  false [] <nil> } {map[] map[]  [] [] {{0}} {         } [] [] [] <nil>}}) %!s(*cache.informerCache=&{0xc00097dc40}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 13:24:40.647086       1 controller.go:185] controller node controllerGroup  controllerKind Node: Starting EventSource source &{{%!s(*v1.Pod=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} {[] [] [] []  <nil> <nil>  map[]   <nil>  false false false <nil> <nil> []   <nil>  [] []  <nil> <nil> [] <nil> <nil> <nil> map[] [] <nil> <nil>} { []      [] <nil> [] []  []}}) %!s(*cache.informerCache=&{0xc00097dc40}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 13:24:40.647538       1 controller.go:185] controller node controllerGroup  controllerKind Node: Starting EventSource source kind source: *v1.Pod
I0906 13:24:40.647827       1 controller.go:193] controller node controllerGroup  controllerKind Node: Starting Controller
I0906 13:24:40.648896       1 controller.go:150] controller endpoints controllerGroup  controllerKind Endpoints: Starting EventSource source &source.Kind{Type:(*v1.Endpoints)(0xc000ebd400), cache:(*cache.informerCache)(0xc000378998), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 13:24:40.649209       1 controller.go:150] controller endpoints controllerGroup  controllerKind Endpoints: Starting EventSource source &source.kindWithCache{kind:source.Kind{Type:(*v1.Endpoints)(0xc000ebd2c0), cache:(*cache.informerCache)(0xc0003786d8), started:(chan error)(nil), startCancel:(func())(nil)}}
I0906 13:24:40.651162       1 init.go:62] init-manifests-controller: process init manifests
I0906 13:24:40.681989       1 controller.go:193] controller ingress controllerGroup networking.k8s.io controllerKind Ingress: Starting Controller
I0906 13:24:40.688197       1 controller.go:227] controller ingress controllerGroup networking.k8s.io controllerKind Ingress: Starting workers worker count 1
I0906 13:24:40.704924       1 controller.go:185] controller persistent-volume-claim controllerGroup  controllerKind PersistentVolumeClaim: Starting EventSource source kind source: *v1.PersistentVolumeClaim
I0906 13:24:40.705487       1 controller.go:185] controller persistent-volume-claim controllerGroup  controllerKind PersistentVolumeClaim: Starting EventSource source &{{%!s(*v1.PersistentVolumeClaim=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} {[] <nil> {map[] map[]}  <nil> <nil> <nil> <nil>} { [] map[] [] map[] <nil>}}) %!s(*cache.informerCache=&{0xc00097dc40}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 13:24:40.705877       1 controller.go:193] controller persistent-volume-claim controllerGroup  controllerKind PersistentVolumeClaim: Starting Controller
I0906 13:24:40.707151       1 controller.go:150] controller ingress controllerGroup networking.k8s.io controllerKind Ingress: Starting EventSource source &source.Kind{Type:(*v1.Ingress)(0xc0009e8f00), cache:(*cache.informerCache)(0xc000378998), started:(chan error)(nil), startCancel:(func())(nil)}
I0906 13:24:40.707454       1 controller.go:150] controller ingress controllerGroup networking.k8s.io controllerKind Ingress: Starting EventSource source &source.kindWithCache{kind:source.Kind{Type:(*v1.Ingress)(0xc0009e8d80), cache:(*cache.informerCache)(0xc0003786d8), started:(chan error)(nil), startCancel:(func())(nil)}}
deployment.apps/coredns configured
I0906 13:24:40.744982       1 controller.go:185] controller networkpolicy controllerGroup networking.k8s.io controllerKind NetworkPolicy: Starting EventSource source kind source: *v1.NetworkPolicy
I0906 13:24:40.745678       1 controller.go:185] controller networkpolicy controllerGroup networking.k8s.io controllerKind NetworkPolicy: Starting EventSource source &{{%!s(*v1.NetworkPolicy=&{{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[] [] []  []} {{map[] []} [] [] []} {[]}}) %!s(*cache.informerCache=&{0xc00097dc40}) %!s(chan error=<nil>) %!s(func()=<nil>)}}
I0906 13:24:40.746563       1 controller.go:193] controller networkpolicy controllerGroup networking.k8s.io controllerKind NetworkPolicy: Starting Controller
service/kube-dns unchanged
I0906 13:24:40.861379       1 start.go:404] CoreDNS configuration from the manifest file applied successfully
I0906 13:24:40.892464       1 controller.go:227] controller event controllerGroup  controllerKind Event: Starting workers worker count 1
I0906 13:24:40.897719       1 controller.go:227] controller kubernetes_default_endpoint controllerGroup  controllerKind Endpoints: Starting workers worker count 1
I0906 13:24:40.901882       1 syncer.go:66] event: coredns-6c54db6fdd-hcfjs-x-kube-system-x-devops.171247c1c43316df: update virtual event kube-system/coredns-6c54db6fdd-hcfjs.171247c1c43316df
I0906 13:24:40.903637       1 controller.go:227] controller persistentvolume controllerGroup  controllerKind PersistentVolume: Starting workers worker count 1
I0906 13:24:40.907455       1 controller.go:117] controller persistentvolume controllerGroup  controllerKind PersistentVolume: persistentVolume klog.ObjectRef{Name:"myvolume", Namespace:""} namespace  name myvolume: reconcileID "5060d8df-5349-443f-80bc-886163816c95": : Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x18843ca]

goroutine 1118 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x1ee7cc0, 0x36bb760})
	/usr/local/go/src/runtime/panic.go:838 +0x207
github.com/loft-sh/vcluster/pkg/controllers/resources/persistentvolumes.(*persistentVolumeSyncer).translateUpdateBackwards(0xc000940f00?, 0xc00064b1f0, 0xc00117b400, 0xc00117bb80, 0x0)
	/vcluster-dev/pkg/controllers/resources/persistentvolumes/translate.go:70 +0x22a
github.com/loft-sh/vcluster/pkg/controllers/resources/persistentvolumes.(*persistentVolumeSyncer).Sync(0x1ec6880?, 0xc00064b1f0, {0x25939f0?, 0xc00117bb80}, {0x25939f0, 0xc00117b400})
	/vcluster-dev/pkg/controllers/resources/persistentvolumes/syncer.go:153 +0x447
github.com/loft-sh/vcluster/pkg/controllers/syncer.(*syncerController).Reconcile(0xc000636930, {0x2580690?, 0xc0005c2a20}, {{{0x0?, 0x10?}, {0xc000b8bc95?, 0x40d9a7?}}})
	/vcluster-dev/pkg/controllers/syncer/syncer.go:109 +0x705
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x25805e8?, {0x2580690?, 0xc0005c2a20?}, {{{0x0?, 0x2106f80?}, {0xc000b8bc95?, 0x4041f4?}}})
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:121 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000953720, {0x25805e8, 0xc0006ba2c0}, {0x1fa4960?, 0xc000f1ea60?})
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:320 +0x33c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000953720, {0x25805e8, 0xc0006ba2c0})
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:273 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:234 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/vcluster-dev/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:230 +0x325

```

The error above doesn't show up on vcluster version 0.11.1, only on 0.11.2.
On 0.11.1 we were able to create PVs.

### Host cluster Kubernetes version

<details>

```console
$ kubectl version
ArgoCD: v2.1.10+de6735c 
Kubectl: v0.21.0
```

</details>


### Host cluster Kubernetes distribution

<details>

```
OKE v1.24.1 (Oracle Cloud Kubernetes, managed)
```

</details>


### vlcuster version

<details>

```console
$ vcluster --version
vcluster version 0.11.2
```

</details>


### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
rancher-k3s:release-v1.23.7-k3s1
```

</details>


### OS and Arch

<details>

```
OS: Oracle Linux 8.6
Arch: amd 64
```
</details>

--- a/pkg/controllers/resources/persistentvolumes/translate.go
+++ b/pkg/controllers/resources/persistentvolumes/translate.go
@@ -67,8 +67,10 @@ func (s *persistentVolumeSyncer) translateUpdateBackwards(ctx *synccontext.SyncC
 		isStorageClassCreatedOnVirtual = equality.Semantic.DeepEqual(storageClassPhysicalName, translatedSpec.StorageClassName)
 
 		// check if claim was created on virtual
-		claimRefPhysicalName := translate.PhysicalName(vPv.Spec.ClaimRef.Name, vPv.Spec.ClaimRef.Namespace)
-		isClaimRefCreatedOnVirtual = equality.Semantic.DeepEqual(claimRefPhysicalName, translatedSpec.ClaimRef.Name)
+		if vPv.Spec.ClaimRef != nil && translatedSpec.ClaimRef != nil {
+			claimRefPhysicalName := translate.PhysicalName(vPv.Spec.ClaimRef.Name, vPv.Spec.ClaimRef.Namespace)
+			isClaimRefCreatedOnVirtual = equality.Semantic.DeepEqual(claimRefPhysicalName, translatedSpec.ClaimRef.Name)
+		}
 	}
 
 	// check storage class. Do not copy the name, if it was created on virtual.

